bus-mapping/src/circuit_input_builder/input_state_ref.rs

```diff
@@ -609,18 +609,7 @@ impl<'a> CircuitInputStateRef<'a> {
             sender_balance_prev,
             sender_balance
         );
-        // If receiver doesn't exist, create it
-        if !receiver_exists && (!value.is_zero() || must_create) {
-            self.push_op_reversible(
-                step,
-                AccountOp {
-                    address: receiver,
-                    field: AccountField::CodeHash,
-                    value: CodeDB::empty_code_hash().to_word(),
-                    value_prev: Word::zero(),
-                },
-            )?;
-        }
+
```

Comment:
It would cause rw inconsistent compared with the original implementation.
Let's say `receiver_exists` is false and `value` is zero, and
1. if `must_create` (or `opcode_is_create`) is false. It works well.
2. but if  `must_create` (or `opcode_is_create`) is true (a.k.a. this if condition holds), it pushes `AccountOp` to tw table. In your changes, you'll miss this account write operation bcs it returns at L624.

I guess that's the reason why light testing was failed.


I guess it doesn't matter but not 100% sure is that the order of rw table will be different in this change.
Before: (receiver, CodeHash) --> (sender, Balance) --> (receiver, Balance)
Now: (sender, Balance) --> (receiver, CodeHash) --> (receiver, Balance)

Reply:
Ah yes, I fixed the begin_tx side but forgot the rest of the places. Let me do a quick fix.